curve25519-dalek/src/field.rs

```diff
@@ -226,6 +242,189 @@ impl FieldElement {
         let t18 = t17.pow2k(50);  // 249..50
         let t19 = &t18 * &t13;  // 249..0
 
+        proof {
+            // Assert the field operation postconditions that the lemma requires
+            pow255_gt_19();  // Prove p() > 0
+
+            // Square operation postconditions (from .square() method ensures clause)
+            assert(as_nat(t0.limbs) % p() == pow(as_nat(self.limbs) as int, 2) as nat % p()) by {};
+            assert(as_nat(t0_sq.limbs) % p() == pow(as_nat(t0.limbs) as int, 2) as nat % p()) by {};
+            assert(as_nat(t1.limbs) % p() == pow(as_nat(t0_sq.limbs) as int, 2) as nat % p()) by {};
+
+            // For mul operations, use lemma to convert from field_mul to direct multiplication
+            assert(as_nat(t2.limbs) % p() == (as_nat(self.limbs) * as_nat(t1.limbs)) % p()) by {
+                lemma_mul_mod_noop_general(
+                    as_nat(self.limbs) as int,
+                    as_nat(t1.limbs) as int,
+                    p() as int,
+                );
+            };
+            assert(as_nat(t3.limbs) % p() == (as_nat(t0.limbs) * as_nat(t2.limbs)) % p()) by {
+                lemma_mul_mod_noop_general(
+                    as_nat(t0.limbs) as int,
+                    as_nat(t2.limbs) as int,
+                    p() as int,
+                );
+            };
+
+            // Use our clean lemma to prove t3 = x^11 and all intermediate steps
+            lemma_pow22501_prove_t3(
+                self.limbs,
+                t0.limbs,
+                t0_sq.limbs,
+                t1.limbs,
+                t2.limbs,
+                t3.limbs,
+            );
+
+            // The lemma proves:
+            // - as_nat(t0_sq.limbs) % p() == pow(base, 4) as nat % p()
+            // - as_nat(t1.limbs) % p() == pow(base, 8) as nat % p()
+            // - as_nat(t2.limbs) % p() == pow(base, 9) as nat % p()
+            // - as_nat(t3.limbs) % p() == pow(base, 11) as nat % p()
+
+            let base = as_nat(self.limbs) as int;
+
+            // Prove t19 = x^(2^250-1) using explicit lemma
+            // Square operation postcondition (t4 = t3^2, from .square() ensures clause)
+            assert(as_nat(t4.limbs) % p() == pow(as_nat(t3.limbs) as int, 2) as nat % p()) by {};
```

Comment:
For this and the following, I'd change `assert(...) by {}` with just `assert(...)` if the `by`-block is empty.